dwapi/datasets.go

```diff
@@ -59,58 +72,93 @@ func (s *DatasetService) CreateOrReplace(owner, id string, body *DatasetReplaceR
 	return s.client.DOI.Delete(owner, datasetid, doi)
 }
 
+// DeleteDOIAssociatedWithVersion deletes a DOI (Digital Object Identifier) associated with a version
+// of a dataset.
 func (s *DatasetService) DeleteDOIAssociatedWithVersion(owner, datasetid, versionid, doi string) (
 	response SuccessResponse, err error) {
 	return s.client.DOI.DeleteAssociatedWithVersion(owner, datasetid, versionid, doi)
 }
 
+// DownloadFile downloads a file within the dataset as originally uploaded.
+//
+// Prefer `Query.ExecuteSQL()` or `Query.ExecuteSPARQL()` for retrieving clean and structured data.
 func (s *DatasetService) DownloadFile(owner, datasetid, filename string) (response io.Reader, err error) {
 	return s.client.File.Download(owner, datasetid, filename)
 }
 
+// DownloadAndSaveFile downloads a file within the dataset as originally uploaded, and saves the results
+// to a file.
+//
+// Prefer `Query.ExecuteSQL()` or `Query.ExecuteSPARQL()` for retrieving clean and structured data.
 func (s *DatasetService) DownloadAndSaveFile(owner, datasetid, filename, path string) (
 	response SuccessResponse, err error) {
 	return s.client.File.DownloadAndSave(owner, datasetid, filename, path)
 }
 
+// Download downloads a .zip file containing all files within a dataset as originally uploaded.
+//
+// Prefer `Query.ExecuteSQL()` or `Query.ExecuteSPARQL()` for retrieving clean and structured data.
 func (s *DatasetService) Download(owner, datasetid, filename string) (response io.Reader, err error) {
 	return s.client.File.DownloadDataset(owner, datasetid)
 }
 
+// DownloadAndSave downloads a .zip file containing all files within a dataset as originally
+// uploaded, and saves the results to a file.
+//
+// Prefer `Query.ExecuteSQL()` or `Query.ExecuteSPARQL()` for retrieving clean and structured data.
 func (s *DatasetService) DownloadAndSave(owner, datasetid, path string) (response SuccessResponse, err error) {
 	return s.client.File.DownloadAndSaveDataset(owner, datasetid, path)
 }
 
+// Liked lists the datasets that the currently authenticated user has liked (bookmarked).
 func (s *DatasetService) Liked() (response []DatasetSummaryResponse, err error) {
 	return s.client.User.DatasetsLiked()
 }
 
+// ListQueries lists the saved queries associated with a dataset.
+//
+// Query definitions will be returned, not the query results. To retrieve the query results
+// use `Query.ExecuteSavedQuery`.
 func (s *DatasetService) ListQueries(owner, datasetid string) (response []QuerySummaryResponse, err error) {
 	return s.client.Query.ListQueriesAssociatedWithDataset(owner, datasetid)
 }
 
+// Owned lists the datasets that the currently authenticated user has access to because they are
+// the owner.
 func (s *DatasetService) Owned() (response []DatasetSummaryResponse, err error) {
 	return s.client.User.DatasetsOwned()
 }
 
+// Retrieve fetches a dataset.
+//
+// The definition of will be returned, not the associated data. Use `Query.ExecuteSQL()`
```

Comment:
"The definition of" what "will be returned"?

Reply:
Lmao, fixed!